### ISSUE TITLE
Nextcloud contacts: no surname

### ISSUE THREAD

**Issue by rubo77** [SimpleMobileTools/Simple-Contacts#965]
If I create a contract in nextcloud and sync it to my Samsung with davX⁵ there is only the first name shown.

**Comment by tibbi** [comment]
does it show in any other app? Arent you filling out the middle name instead of surname?

**Comment by rubo77** [comment]
Yes, in the standard Samsung contact app on my S10+ it shows correct.

The problem seems, that the title of a contact is a free text field for whatever you want and the fields Name and surname are only filled, if you edit them by hand in "detaillierter Name"

**Comment by michaelof** [comment]
@rubo77 Hi, what you're describing is a generic issue with Nextcloud in combination with SimpleContacts. Nextcloud supports the full name concept, whatever you enter, in one field, detailed name only on demand. SimpleContacts simply doesn't, detailed name only.
To use Nextcloud with SimpleContacts, what I'm doing happily for many years now, it's worth to look in detail how Nextcloud handles these names:
Starting in Nextcloud (NC)
- When you create a fullname only, only one name = no blanks, you'll have no detailed name in NC, at first. Once synched with SimpleContacts (SC), detailed name is added(!) to cardDAV item, only first name filled, as you described.
- when you create a fullname only, but two name parts, separated by blank/ white space, same autom. add of detailed name as before, but First and Last are filled in detailed name, after synch.
- when you start with detailed name in NC, synch is more or less done fine, more or less = some pitfalls with prefixes, like "Dr. Dr. med. Herbert Mustermann", order mixed up, but out of scope here :)
- NOT completely comprehensible to me is NC's way of dealing with changes of detailed name parts back to fullname. If you start with detailed name, fullname is set dynamically. Up to whatever event that "triggers" fullname to be persistant. From then on, changes to detailed name will not change fullname, anymore.
- vice versa: changes to fullname with ex. detailed name will not change detailed name in any way
- but: "reset" possible: Erase fullname will trigger NC to regenerate it based on detailed name!

But another big BUT: Whenever you make whatever change to a contact in SC, previous fullname will be DESTROYED in NC by next synch!!! Simple reason is, as said in the beginning, that fullname is not in scope for SC, so just overwritten in case of cardDAV synch from SC to NC.

Knowing all this, there are several ways how to deal with it. My personal way, as I don't like NC's default to set initial or resetted fullname to "first last", is a (perl) script which sets fullname to what I want, server side, NC. FYI I prefer fullname to be set as "last, [prefixes] first". Reason behind is, that several contacts using devices, esp. cars via Bluetooth, also have their specialities how they're dealing with the fullname issue :) And prefer to find Dr. Dr. med Herbert Mustermann by starting with an M when driving...

**Comment by rubo77** [comment]
Why not simply implement it the same way Samsung contacts handles the problems?
